install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,110 @@
+====
+
+[NOTE]
+====
+External attacher must run also for CSI drivers that don't support 3rd party attach/detach operations. External attacher will not issue any `ControllerPublish` or `ControllerUnpublish` to CSI driver, however it still must run to implement necessary {product-title} attachment API.
+====
+
+=== External CSI provisioner deployment
+
+External CSI provisioner is a Deployment that deploys single or several pods with two containers:
+
+  * External CSI provisioner container itself that translates `provision` and `delete` calls from {product-title} to respective `CreateVolume` and `DeleteVolume` calls to CSI driver.
+  * CSI driver container.
+
+CSI provisioner container talks to CSI driver container using Unix Domain Sockets, ensuring that no CSI communication leaves the pod. CSI driver is not accessible from outside of the pod at all.
+
+Deployment can be used to scale up/down number of provisioners.
+
+[NOTE]
+====
+`provision` and `delete` operations typically require CSI driver to use credentials to the storage backend. It is recommended to run external provisioner pods on xref:../../install_config/install/advanced_install.adoc#configuring-dedicated-infrastructure-nodes[infrastructure nodes] so the credentials can never leak to user processes even in case of catastrophic security breach on a node.
+====
+
+=== CSI driver DaemonSet
+
+Finally, CSI driver DaemonSet runs a pod on every node that allows {product-title} to mount storage provided by CSI driver to the node and use it in user workloads (Pods) as persistent volumes. The pod with CSI driver contains these containers:
+
+* CSI driver registrar, that registers the CSI driver into `openshift-node` service running on the node. `openshift-node` process running on the node then directly talks to CSI driver using Unix Domain Socket available on the node.
+* CSI driver itself.
+
+CSI driver deployed on the node should have as least credentials to the storage backend as possible. {product-title} will use just Node plugin set of CSI calls such as `NodePublish`/`NodeUnpublish` and `NodeStage`/`NodeUnstage` (if implemented).
+
+== Example deployment
+
+Since {product-title} does not ship any CSI driver, this example shows how to deploy community driver for OpenStack Cinder in {product-title}.
+
+TODO: example yaml file that uses OSE images + shell commands how to install it.
+
+== Dynamic provisioning
+
+Dynamic provisioning of persistent storage depends on capabilities of CSI driver and underlying storage backend. Provider of a CSI driver should document how to create StorageClass in {product-title} and what parameters are available for configuration.
```

Comment:
s/depends on capabilities/of CSI driver/depends on the capabilities of the CSI driver

s/Provider of a CSI driver/The provider of the CSI driver